.github/workflows/ci.yml

```diff
@@ -1103,110 +1103,6 @@ ${{ hashFiles('.pre-commit-config.yaml') }}"
           path: /tmp/kind_logs_*
           retention-days: 7
 
-  push-prod-images-to-github-registry:
```

Comment:
This is now merged into single job after constraints are pushed, and the build happens using those new constraints, so we are going to have new tested constraints always reflected in the latest cache images. 

I could also potentially separate out this change I think If you think it is too much in one PR.